Customer.java

```diff
@@ -3,59 +3,73 @@
 
 import Client,Cinema,Movie;
 
-public class Customer implements Client {
-    private Ticket[] bookings;  //previous bookings made by customer given by tickets.
-    private boolean auth = false;  //authenticatiuon indicator. Whether customer is authenticated or not. All methods must check for authentication.
+public class Customer implements Client, Serializable{
+
+    private int age;
+    private String name;
+    private boolean auth = false;  //authentication indicator. Whether customer is authenticated or not. All methods must check for authentication.
     public String username;
+    public String password;
 
-    public boolean login(String password, Database db) {
-        if (!db.user_accounts.containsKey(this.username)) return false;
+    //Constructor for Customer object.
+    public Customer(String name, int age) {
+        this.name = name;
+        this.age = age;
+    }
 
-        else if (db.user_accounts.get(this.username) != password) return false;
-        this.auth = true;
-        System.out.println("Authenticated succesfully");
-        bookings = new Ticket[100];
-        return true;
+    public Customer(String name, int age, String username, String password) {
+        this.name = name;
```

Comment:
Seems like you've defined the constructor for Customer twice. While there's no technical problem in defining multiple constructors, it seems that the first one is really just a subset of the second constructor in terms of the inputs taken. So maybe we can delete the first `Customer(String name, int age)` since that's just a subset of the second one

Reply:
Correct me if I am wrong since I'm still a little confused with the constructors. 
The reason why I have 2 different constructor is because one is for when we first get the customer name and age in the main function so at that instance, the user does not have a username and password yet (For this reason, I created 2 constructor, 1 for before username and password is made and the other for after they are created) 

If I remove the constructor with 2 parameters, when creating the object in the initial main function, do we pass in Customer(name, age, null ,null)? Will that work?

Reply:
So here's the thing, we will always have the email and password of Customer object while cerating one. Why? Because Customers are going to be pre-built using CRUD. Therefore, you need to have a CRUD method that will read in information from the CustomerDB.txt database, and create all the Customer objects(with their required attribs) in the beginning itself.